pkg/splunk/controller/statefulset.go

```diff
@@ -85,8 +86,102 @@ func ApplyStatefulSet(c splcommon.ControllerClient, revised *appsv1.StatefulSet)
 	return splcommon.PhaseReady, nil
 }
 
+// UpdatePodRevisionHash updates the controller-revision-hash label on pods.
+func UpdatePodRevisionHash(c splcommon.ControllerClient, statefulSet *appsv1.StatefulSet, readyReplicas int32) error {
+	scopedLog := log.WithName("updatePodRevisionHash").WithValues(
+		"name", statefulSet.GetObjectMeta().GetName(),
+		"namespace", statefulSet.GetObjectMeta().GetNamespace())
+
+	namespacedName := types.NamespacedName{Namespace: statefulSet.GetNamespace(), Name: statefulSet.GetName()}
+	var current appsv1.StatefulSet
+
+	err := c.Get(context.TODO(), namespacedName, &current)
+	if err != nil {
+		scopedLog.Error(err, "Unable to Get statefulset", "statefulset", statefulSet.GetName())
+	}
+
+	for n := readyReplicas - 1; n >= 0; n-- {
+		// get Pod
+		podName := fmt.Sprintf("%s-%d", current.GetName(), n)
+		namespacedName := types.NamespacedName{Namespace: current.GetNamespace(), Name: podName}
+		var pod corev1.Pod
+		err := c.Get(context.TODO(), namespacedName, &pod)
+		if err != nil {
+			scopedLog.Error(err, "Unable to find Pod", "podName", podName)
+			return err
+		}
+		if pod.Status.Phase != corev1.PodRunning || len(pod.Status.ContainerStatuses) == 0 || pod.Status.ContainerStatuses[0].Ready != true {
+			scopedLog.Error(err, "Waiting for Pod to become ready", "podName", podName)
+			return err
+		}
+
+		labels := pod.GetLabels()
+		revisionHash := labels["controller-revision-hash"]
+		// update pod controller revision hash
+		if current.Status.UpdateRevision != "" && current.Status.UpdateRevision != revisionHash {
+			// update controller-revision-hash label for pod with statefulset update revision
+			labels["controller-revision-hash"] = current.Status.UpdateRevision
```

Comment:
This "controller-revision-hash" param is set to "v0" at pod bringup then gets updated when a scale-up/down event occurs?  Where is this "controller-revision-hash" updated other than here?  Is this typically done only when the pod is spun up?

Reply:
I dont think so it is "v0" during bring up. I checked in my local env and it has definite hash value,probably similar to updateRevision of statefulset. And yes it gets updated during any update to statefulset, not necessarily scale up/down only( Check `ApplyStatefulSet`).

Reply:
nm, I was looking at the Tester code.